### DISCUSSION TITLE
Self-hosted runners can't be used when Actions Service is unstable

### DISCUSSION THREAD

**Discussion by AllanOricil** [community/community#55060] [Actions]
### Select Topic Area

Product Feedback

### Body

When the actions service is unstable, customers can't run their jobs, even when using their own "runners" (self-hosted).
This issue is rare, but it happens. What if customers could run their own "actions" service?

We could develop our own Actions Service, but that would take time. So what if Github makes it public, and allows us to connect it to the remote repository? 

Github (repositories) <-> Self-hosted Actions Service <-> Self-hosted Runners 

Github could also allow customers to change the route of the "Actions" tab so that it could redirect us to our own service.

<img width="148" alt="image" src="https://github.com/community/community/assets/55927613/7baa692e-790d-44ee-b964-1a58fe6a17b6">

**Comment by aspyropoulos** [comment]
Can someone from the MS can finally fix github services once and for all so that we can work at some point? Last month has been immensely annoying these service disruptions incuring to us lots of customer frustration.

**Comment by ChristopherHX** [comment]
You can self-host the Actions Service for self-hosted runners together with GitHub Enterprise Server https://docs.github.com/en/enterprise-server@3.8/admin/overview/about-github-enterprise-server.
However it more expensive than GitHub Free.

**Reply by AllanOricil** [reply]
It would be nice to have the "official" one decoupled from the enterprise server. It would be even nicer if there was some kind of "configure you own actions service button" somewhere in Github that would guide a user to setup it on its own infrastructure for jobs. Not only the actions service, but also runners.

**Comment by airtower-luna** [comment]
There's a (partial) implementation for running Actions workflows locally: https://github.com/nektos/act

Not quite the same thing as an open Actions control server, but might be helpful.

**Comment by ChristopherHX** [comment]
> We could develop our own Actions Service

I did this already and yes you can technically use it together with github.com including their official self-hosted runners, but
I don't promise
- stability
- performance
- beeing able to run it at scale
- for 3 months without restarts

https://github.com/ChristopherHX/runner.server
- it is not only a cli tools
- it is a real aspnetcore server
- Manual setup needed to make it fully functional

**Reply by AllanOricil** [reply]
jesus @ChristopherHX 
sometimes It feels like you are a stalker. You are always commenting on my stuff.

**Reply by AllanOricil** [reply]
I liked what you did. Thanks for sharing. 
It would be nice if we could bind our own actions service to Github's UI. For instance, I would like to click on the "Actions" tab and be redirected to my own actions service. And I would also want to keep the Github's look and feel.

**Reply by ChristopherHX** [reply]
I fully agree, the missing GitHub UI integration and the unstyled UI are big pain points. My biggest pain point is missing free GitHub Hosted runners, so I just wait for GitHub to fix their stuff for my public projects.

> You are always commenting on my stuff.

I monitor actions/runner and this forum to improve my GitHub Actions Emulator and keep up with (undocumented) features. If you comment there, it is very likly I would read your comments and might answer..
